src/components/Modal/Modal.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+
+import './Modal.scss';
+
+const CN = 'modal';
+
+class Modal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isTransition: false
+    };
+
+    this.modalRef = React.createRef();
+    this.setTransition = this.setTransition.bind(this);
+    this.hideModalOnBackdrop = this.hideModalOnBackdrop.bind(this);
+    this.hideModalOnEscape = this.hideModalOnEscape.bind(this);
+    this.removeModalOnTransitionEnd = this.removeModalOnTransitionEnd.bind(this);
+  }
+
+  componentDidMount() {
+    document.body.classList.add('remove-scroll');
+    this.modalRef.current.focus();
+  }
+
+  componentWillUnmount() {
+    document.body.classList.remove('remove-scroll');
+    this.modalRef.current.removeEventListener('transitionEnd', this.removeModal);
+  }
+
+  setTransition = () => {
+    this.setState({
+      isTransition: true
+    });
+  };
+
+  hideModalOnBackdrop = (event) => {
+    event.target.classList.contains(CN) && this.setTransition();
+  };
+
+  hideModalOnEscape = (event) => {
+    const escapeKeyCode = 27;
+
+    event.keyCode === escapeKeyCode && this.setTransition();
+  };
+
+  removeModalOnTransitionEnd(e) {
+    const { removeModal } = this.props;
+
+    if (e.target.classList.contains(CN)) {
+      this.setState({
+        isTransition: false
+      });
+
+      removeModal();
+    }
+  }
+
+  render() {
+    const { children, className } = this.props;
+    const { isTransition } = this.state;
+
+    return (
+      <div
+        tabIndex="-1"
+        ref={this.modalRef}
+        className={cx(CN, { [`${CN}--transition`]: isTransition })}
+        onClick={this.hideModalOnBackdrop}
+        onKeyDown={this.hideModalOnEscape}
+        onTransitionEnd={this.removeModalOnTransitionEnd}
+      >
+        <div
+          className={cx(`${CN}__window`, className)}
+        >
+          {children}
+          <span
```

Comment:
`Type '{ children: Element; className: string; role: string; label: string; tabIndex: number; onClick: () => void; }' is not assignable to type 'DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>'.
  Property 'label' does not exist on type 'DetailedHTMLProps<HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>'.ts(2322)`

Reply:
Please take into account invalid properties for plain HTMLElements